stories/NumberInput.stories.tsx

```diff
@@ -0,0 +1,101 @@
+
+import React from 'react';
+
+import { Story } from '@storybook/react';
+
+import { NumberInput } from '../src/components/atoms';
+import { NumberInputProps } from '../src/components/atoms/NumberInput/NumberInput';
+
+
+export default {
+  title: 'NumberInput',
+  component: NumberInput
+};
+
+
+const Template: Story<NumberInputProps> = (args) => {
+  const [ value, setValue ] = React.useState('100');
+
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  return (
+    <>
+      <NumberInput
+        {...args}
+        value={value}
+        onChange={onChange}
+      />
+    </>
+  );
+};
+
+export const Primary = Template.bind({});
+Primary.args = {
+  placeholder: 0
+};
+
+export const Warning = Template.bind({});
+Warning.args = {
+  value: 100,
+  variant: 'warning'
+};
+
+
+export const Error = Template.bind({});
+Error.args = {
+  value: 100,
+  variant: 'error'
+};
+
+export const DisabledSpecialChar = Template.bind({});
+DisabledSpecialChar.args = {
+  value: 100,
+  allowSpecialCharacters: true
+};
+
+export const DisableDecimal = Template.bind({});
+DisableDecimal.args = {
+  value: 100,
+  disableDecimal: true
+};
+
+export const RupeeSymbol = Template.bind({});
```

Comment:
are these necessary?, can't we force the user to use NumberFormater instead?

Reply:
we can discuss this

Reply:
This is an exapmle where we show rupee symbol as prefix element

Reply:
We don't have Number formatter in ui-toolkit